File: frontend/features/workspace/pets/pet-profile.ts
import type { PetProfile } from "@/app/lib/api";
import { petAdvisors } from "./pet-state.mjs";

export const petDefaults: PetProfile[] = [
  { slot: "LEAN", name: "차근", animal: "turtle", color: "sage", accessory: "none", tone: "WARM", valuePriority: "BALANCED", deliveryPriority: "SPEED", scopePriority: "CAUTIOUS" },
  { slot: "RECOMMENDED", name: "또렷", animal: "owl", color: "lavender", accessory: "none", tone: "FORMAL", valuePriority: "BALANCED", deliveryPriority: "QUALITY", scopePriority: "BALANCED" },
  { slot: "EXPANDED", name: "든든", animal: "cat", color: "peach", accessory: "none", tone: "DIRECT", valuePriority: "PROFIT", deliveryPriority: "BALANCED", scopePriority: "EXPLORATORY" }
];
export const petColors = { sage: "세이지", lavender: "라벤더", peach: "살구", sky: "하늘", rose: "장미", ink: "먹색" };
export const preferenceLabels: Record<string, string> = { PROFIT: "수익 우선", RELATIONSHIP: "관계 우선", BALANCED: "균형", SPEED: "빠른 납품", QUALITY: "완성도 우선", CAUTIOUS: "보수적 범위", EXPLORATORY: "도전적 제안" };
export function advisorsWithProfiles(profiles?: PetProfile[]) {
  return petAdvisors.map(advisor => {
    const profile = profiles?.find(pet => pet.slot === advisor.scenario) ?? petDefaults.find(pet => pet.slot === advisor.scenario)!;
    return { ...advisor, name: profile.name, profile, priority: [profile.valuePriority, profile.deliveryPriority, profile.scopePriority].map(value => preferenceLabels[value]).join(" · ") };
  });
}

File: backend/src/main/java/com/freelanceops/backend/domain/agentrun/dto/PetProfile.java
package com.freelanceops.backend.domain.agentrun.dto;

import jakarta.validation.constraints.*;
import java.util.List;

public record PetProfile(
    @NotNull @Pattern(regexp = "LEAN|RECOMMENDED|EXPANDED") String slot,
    @NotBlank @Size(max = 20) @Pattern(regexp = "[\\p{L}\\p{N} _-]+") String name,
    @NotNull @Pattern(regexp = "turtle|owl|cat") String animal,
    @NotNull @Pattern(regexp = "sage|lavender|peach|sky|rose|ink") String color,
    @NotNull @Pattern(regexp = "none|glasses|scarf|star") String accessory,
    @NotNull @Pattern(regexp = "WARM|DIRECT|FORMAL") String tone,
    @NotNull @Pattern(regexp = "PROFIT|BALANCED|RELATIONSHIP") String valuePriority,
    @NotNull @Pattern(regexp = "SPEED|BALANCED|QUALITY") String deliveryPriority,
    @NotNull @Pattern(regexp = "CAUTIOUS|BALANCED|EXPLORATORY") String scopePriority
) {
    public static List<PetProfile> defaults() {
        return List.of(
            new PetProfile("LEAN", "차근", "turtle", "sage", "none", "WARM", "BALANCED", "SPEED", "CAUTIOUS"),
            new PetProfile("RECOMMENDED", "또렷", "owl", "lavender", "none", "FORMAL", "BALANCED", "QUALITY", "BALANCED"),
            new PetProfile("EXPANDED", "든든", "cat", "peach", "none", "DIRECT", "PROFIT", "BALANCED", "EXPLORATORY")
        );
    }
}
